dev_guide/builds/triggering_builds.adoc

```diff
@@ -90,11 +93,97 @@ $ curl -H "X-GitHub-Event: push" -H "Content-Type: application/json" -k -X POST
 signed certificate.
 
+[[gitlab-webhooks]]
+=== GitLab Webhooks
+
+https://docs.gitlab.com/ce/user/project/integrations/webhooks.html[GitLab webhooks]
+handle the call made by GitLab when a repository is updated. As with the GitHub
+triggers, you must specify a `secret`. The following example is
+a trigger definition YAML within the `BuildConfig`:
+
+[source,yaml]
+----
+type: "GitLab"
+gitlab:
+  secret: "secret101"
+----
+
+The payload URL is returned as the GitLab Webhook URL by the `oc describe` command
+(see xref:describe-buildconfig[Displaying Webhook URLs]), and is structured as follows:
+
+----
+http://<openshift_api_host:port>/oapi/v1/namespaces/<namespace>/buildconfigs/<name>/webhooks/<secret>/gitlab
+----
+
+To configure a GitLab Webhook:
+
+. Describe the build configuration to get the webhook URL:
++
+----
+$ oc describe bc <name>
+----
+. Copy the webhook URL.
+. Follow the https://docs.gitlab.com/ce/user/project/integrations/webhooks.html#webhooks[GitLab setup instructions]
+to paste the webhook URL into your GitLab repository settings.
+
+Given a file containing a valid JSON payload, you can manually trigger the
+webhook via `curl`:
+
+----
+$ curl -H "X-GitLab-Event: Push Hook" -H "Content-Type: application/json" -k -X POST --data-binary @gitlub_payload_file.json https://<openshift_api_host:port>/oapi/v1/namespaces/<namespace>/buildconfigs/<name>/webhooks/<secret>/gitlab
+----
+
+The `-k` argument is only necessary if your API server does not have a properly
+signed certificate.
+
+[[bitbucket-webhooks]]
+=== Bitbucket Webhooks
+
+https://confluence.atlassian.com/bitbucket/manage-webhooks-735643732.html[Bitbucket webhooks]
+handle the call made by Bitbucket when a repository is updated. Similar to the
+previous triggers, you must specify a `secret`. The following example is a
+trigger definition YAML within the `BuildConfig`:
+
+[source,yaml]
+----
+type: "Bitbucket"
+bitbucket:
+  secret: "secret101"
+----
+
+The payload URL is returned as the Bitbucket Webhook URL by the `oc describe` command
+(see xref:describe-buildconfig[Displaying Webhook URLs]), and is structured as follows:
+
+----
+http://<openshift_api_host:port>/oapi/v1/namespaces/<namespace>/buildconfigs/<name>/webhooks/<secret>/bitbucket
+----
+
+To configure a Bitbucket Webhook:
+
+. Describe the build configuration to get the webhook URL:
++
+----
+$ oc describe bc <name>
+----
+. Copy the webhook URL.
+. Follow the https://confluence.atlassian.com/bitbucket/manage-webhooks-735643732.html[Bitbucket setup instructions]
+to paste the webhook URL into your Bitbucket repository settings.
+
+Given a file containing a valid JSON payload, you can manually trigger the
+webhook via `curl`:
+
+----
+$ curl -H "X-Event-Key: repo:push" -H "Content-Type: application/json" -k -X POST --data-binary @gitlub_payload_file.json https://<openshift_api_host:port>/oapi/v1/namespaces/<namespace>/buildconfigs/<name>/webhooks/<secret>/bitbucket
```

Comment:
s/gitlub_payload_file/bitbucket_payload_file/

Or even better -- we can use similar `payload_file.json` name in all examples.